tests/ast/nodes/test_evaluate_binop_int.py

```diff
@@ -0,0 +1,135 @@
+from hypothesis import (
+    example,
+    given,
+    settings,
+    strategies as st,
+)
+import pytest
+
+from vyper import (
+    ast as vy_ast,
+)
+from vyper.exceptions import (
+    ZeroDivisionException,
+)
+
+
+@pytest.mark.fuzzing
+@settings(deadline=500)
+@given(
+    left=st.integers(min_value=-(2 ** 32), max_value=2 ** 32),
+    right=st.integers(min_value=-(2 ** 32), max_value=2 ** 32),
+)
+@pytest.mark.parametrize("op", "+-*/%")
+def test_binop_int128(get_contract, assert_tx_failed, op, left, right):
+    source = f"""
+@public
+def foo(a: int128, b: int128) -> int128:
+    return a {op} b
+    """
+    contract = get_contract(source)
+
+    vyper_ast = vy_ast.parse_to_ast(f"{left} {op} {right}")
+    old_node = vyper_ast.body[0].value
+    try:
+        new_node = old_node.evaluate()
+        is_valid = True
+    except ZeroDivisionException:
+        is_valid = False
+
+    if is_valid:
+        assert contract.foo(left, right) == new_node.value
+    else:
+        assert_tx_failed(lambda: contract.foo(left, right))
+
+
+@pytest.mark.fuzzing
+@settings(deadline=500)
+@given(
+    left=st.integers(min_value=0, max_value=2 ** 64),
+    right=st.integers(min_value=0, max_value=2 ** 64),
+)
```

Comment:
`uint64_st` from above

```suggestion
@given(left=uint64_st, right=uint64_st)
```